warehouse/query-core/src/main/java/datawave/query/jexl/functions/EvaluationPhaseFilterFunctions.java

```diff
@@ -58,32 +56,30 @@ public static boolean occurrence(Iterable<?> values, int count) {
     }
     
     private static int getSizeOf(Iterable<?> iterable) {
-        Map<String,Integer> countMap = Maps.newHashMap();
-        int count = 0;
         if (iterable != null) {
-            // additional datatypes may have been added to this field to assist with evaluation and for use as
-            // a 'marker' for some other condition. We care only about how many were in the original so
-            // bin them all up to count what was really there in the beginning
+            int sourcedFromEvent = 0;
+            int sourcedFromIndex = 0;
```

Comment:
I think I need a third category here for anything that is not a ValueTuple.  I am not sure this is needed but it seems odd to pass in an iterable of non ValueTuple objects and come back with a size of 0.  So if sourcedFromEvent is non-zero return that, else if sourceFromIndex is non-zero then return that , else return the number in the 3rd category..

Reply:
This is now added.